conjure/src/main/java/com/palantir/conjure/cli/ConjureCli.java

```diff
@@ -43,20 +49,63 @@
             .setSerializationInclusion(JsonInclude.Include.NON_ABSENT);
 
     public static void main(String[] args) {
-        System.exit(new CommandLine(new ConjureCli()).execute(args));
+        System.exit(prepareCommand().execute(args));
+    }
+
+    @VisibleForTesting
+    static CommandLine prepareCommand() {
+        return new CommandLine(new ConjureCli()).setExecutionExceptionHandler(new ExceptionHandler());
+    }
+
+    public static final class ExceptionHandler implements IExecutionExceptionHandler {
+        @Override
+        public int handleExecutionException(Exception ex, CommandLine commandLine, ParseResult parseResult)
+                throws Exception {
+            if (ex == null) {
+                return 0;
+            }
+            // For known internal errors, continue, as they're signaling errors on the input, not in the code
+            Throwable rootCause = Throwables.getRootCause(ex);
+            if (!(rootCause instanceof ConjureException)
+                    && !(rootCause instanceof ParseException)
+                    && !(rootCause instanceof CyclicImportException)) {
+                throw ex;
+            }
+            if (!(commandLine.getCommand() instanceof ConjureCliCommand)) {
+                throw ex;
+            }
+            ConjureCliCommand cmd = commandLine.getCommand();
+            if (cmd.isVerbose()) {
+                // If the command is verbose, print the full stack trace
+                throw ex;
+            }
+
+            // Unpack errors; we don't care about where in the code the error comes from: the issue is in the supplied
+            // conjure code. The stack doesn't help.
+            Throwable curr = ex;
+            while (curr != null) {
+                commandLine.getErr().println(curr.getMessage());
+                curr = curr.getCause();
```

Comment:
Probably want to track using an identity hash set here, otherwise this can loop infinitely.

Reply:
Done